internal/controllers/rest_controller.go

```diff
@@ -32,32 +36,44 @@ func NewRESTController(
 	configuration configurations.RESTControllerConfiguration,
 	oidcService services.OIDCService,
 	credentials []handlers.CredentialsHandler,
+	provisionTokenService services.ProvisionTokens,
 ) RESTController {
 	return RESTController{
-		configuration: configuration,
-		oidcService:   oidcService,
-		credentials:   credentials,
-		logger:        logger,
+		configuration:         configuration,
+		oidcService:           oidcService,
+		credentials:           credentials,
+		provisionTokenService: provisionTokenService,
+		logger:                logger,
 	}
 }
 
 // Boot initializes creating some routes.
 func (rc RESTController) Boot(s *server.Server) {
 	s.Router.Use(rc.skipURLsMiddleware())
 
-	s.Router.GET(rc.configuration.AuthServiceURL, gin.WrapF(rc.CheckForAuthentication))
+	s.Router.GET(rc.configuration.AuthServiceURL, rc.CheckForAuthentication)
 	s.Router.GET(rc.configuration.OIDCCallbackURL, gin.WrapF(rc.OIDCProviderCallback))
 	s.Router.GET(rc.configuration.UserInfoURL, gin.WrapF(rc.UserInfo))
 	s.Router.POST(rc.configuration.LogoutURL, gin.WrapF(rc.Logout))
 
-	s.Router.Any("/:forward", gin.WrapF(rc.CheckForAuthentication))
-	s.Router.Any("/:forward/*any", gin.WrapF(rc.CheckForAuthentication))
+	profileTokensURL := "/auth/profiles/:uuid/tokens"
+	s.Router.GET(profileTokensURL, rc.CheckForAuthentication, rc.GetToken)
```

Comment:
this method should belong to the group with the token uid in the path, so:

`/auth/profiles/:uuid/tokens/:tokenUUID`

Reply:
nice catch